### ISSUE TITLE
Add `concat_slice`

### ISSUE THREAD

**Issue by oberien** [oberien/str-concat#6]
See earlier discussion in #3

**Comment by 197g** [comment]
Two possibilities:

1. Add this for slices `[T]` only. This is fairly easy (well, ok, ZSTs are interesting #5).
2. Define a trait to allow adding this for more types. Sketch:
  
  ```
  pub trait Concat {
      type Value;
      fn into_slice(&self) -> &[Value];
      // SAFETY: must only be called with (concatentations of) slices obtained from `into_slice`.
      unsafe fn from_slice(&self) -> &Self;
  }
  
  impl<T> Concat for [T] {  /* trivial */ }
  ```

**Comment by 197g** [comment]
I'd simply do 1. for now as it will be an implementation details that is surely used in 2.

**Comment by oberien** [comment]
Fixed by #7
